fastapi_security/gendigest.py

```diff
@@ -0,0 +1,133 @@
+    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+    ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+    LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+    CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+    """
+    if not isinstance(import_str, str):
+        return import_str
+
+    module_str, _, attrs_str = import_str.partition(":")
+    if not module_str or not attrs_str:
+        message = (
+            'Import string "{import_str}" must be in format "<module>:<attribute>".'
+        )
+        raise ValueError(message.format(import_str=import_str))
+
+    try:
+        module = importlib.import_module(module_str)
+    except ImportError as exc:
+        if exc.name != module_str:
+            raise exc from None
+        message = 'Could not import module "{module_str}".'
+        raise ValueError(message.format(module_str=module_str))
+
+    instance = module
+    try:
+        for attr_str in attrs_str.split("."):
+            instance = getattr(instance, attr_str)
+    except AttributeError:
+        message = 'Attribute "{attrs_str}" not found in module "{module_str}".'
+        raise ValueError(
+            message.format(attrs_str=attrs_str, module_str=module_str)
+        )
+
+    return instance
+
+
+parser = argparse.ArgumentParser(
+    description=_wrap_paragraphs(__doc__),
+    formatter_class=argparse.RawDescriptionHelpFormatter,
+)
+parser.add_argument('fastapi_security_obj')
+
+
+obj = FastAPISecurity()
+obj.init_basic_auth_with_digest('salt123', [])
+
+
+def main():
+    args = parser.parse_args()
+
+    fastapi_security_obj = import_from_string(args.fastapi_security_obj)
+    if callable(fastapi_security_obj):
+        instance = fastapi_security_obj()
+    elif isinstance(fastapi_security_obj, FastAPISecurity):
+        instance = fastapi_security_obj
+    else:
+        print("Cannot generate digest: ", args.fastapi_security_obj,
+              "must point to a FastAPISecurity object or a function returning one",
+              file=sys.error)
+        sys.exit(1)
+
+    password = getpass(prompt='Password: ')
+    password_confirmation = getpass(prompt='Confirm password: ')
+
+    if password != password_confirmation:
+        print("Cannot generate digest: passwords don't match", file=sys.stderr)
+        sys.exit(1)
+
+    print(instance.basic_auth_with_digest.generate_digest(password))
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
There's a big chance that the FastAPISecurity has not been initialized after module import (at least that's how I've been using it). I would prefer if we go with a simpler solution that accepts `salt` as an input argument.

Also I would rename this module to `cli` and make it available on installs as:

```
[tool.poetry.scripts]
fastapi-security = 'fastapi_security.cli:main'
```

Then `gendigest` would be a subcommand (The only one that would exist for the time being). You can check out [this file](https://github.com/jmagnusson/netsuite/blob/main/netsuite/cli/main.py) for an example.

Reply:
> There's a big chance that the FastAPISecurity has not been initialized after module import (at least that's how I've been using it). I would prefer if we go with a simpler solution that accepts salt as an input argument.

This mimics the way asgi applications are configured: they app may not be configured after module import, but there has to be a module containing a variable a configured app or a function returning a configured app. I figured that would be easier to use, because that way command line params don't depend on salt value, which has to be passed to the app somehow anyway, but I can add `salt` as an argument, no prob.

Re: subcommand/script: sure